src/core/ai_circuit_breaker.py

```
@@ -0,0 +1,53 @@
+class CircuitState:
+    failures: int = 0
+    last_failure: Optional[datetime] = None
+    is_open: bool = False
+    opened_at: Optional[datetime] = None
+
+class AICircuitBreaker:
+    def __init__(self, failure_threshold: int = 5, reset_timeout: timedelta = timedelta(minutes=5)):
+        self.failure_threshold = failure_threshold
+        self.reset_timeout = reset_timeout
+        self.circuits: Dict[str, CircuitState] = {}
+        
```

Ruff (0.8.2)

Blank line contains whitespace

Remove whitespace from blank line

(W293)